user/Makefile

```diff
@@ -2,20 +2,34 @@
 
 include ../Makefile.common
 
-SUBDIRS = $(shell find -mindepth 1 -maxdepth 1 -type d -not -name sysroot)
+PROGDIRS = $(shell find -mindepth 1 -maxdepth 1 -type d)
+PROGDIRS-inst = $(PROGDIRS:%=%.inst)
 
-.PHONY: clean sysroot all $(SUBDIRS)
+.PHONY: clean all $(PROGDIRS)
 
-all: $(SUBDIRS)
+all: $(PROGDIRS) install
 
-$(SUBDIRS): sysroot
+install: $(PROGDIRS-inst)
+
+$(PROGDIRS):
+	@echo "Maybe compiling $@..."
```

Comment:
Please keep make messages format consistent, i.e. `[BUILD] $@` and `[SYSROOT] $@`.